src/windows.rs

```diff
@@ -528,42 +526,45 @@ pub fn main() -> Result<()> {
             }
         }
         ExecutionMode::Open => {
-            let config = read_config()?;
+            let Configuration {
+                mut browser_path,
+                custom_osu_path,
+            } = read_config()?;
+
             let client = reqwest::blocking::Client::new();
 
-            let mut found = false;
-            let mut path = config.custom_osu_path.unwrap_or("N/A".into());
-            if path == "N/A".to_string() || path.is_empty() {
-                let osu_path = get_exe_path("osu!.exe").unwrap_or_default();
-                if osu_path.exists() {
-                    found = true;
-                    path = osu_path.to_str().unwrap().to_string();
-                } else {
-                    if let Some(local_app_data) = get_local_app_data_path() {
-                        let default_osu_path = local_app_data.join("osu!").join("osu!.exe");
-                        if default_osu_path.exists() {
-                            found = true;
-                            path = default_osu_path.to_str().unwrap().to_string();
-                        }
+            let osu_path = match custom_osu_path {
+                Some(path) => Some(path),
+                None => {
+                    if let Some(osu_path) = get_exe_path("osu!.exe").filter(|path| path.exists()) {
+                        Some(osu_path)
+                    } else if let Some(local_app_data) = get_local_app_data_path() {
+                        let mut default_osu_path = local_app_data;
+                        default_osu_path.push("osu!/osu!.exe");
```

Comment:
```suggestion
                        let default_osu_path = local_app_data.join("osu!/osu!.exe");
```

Reply:
I think this is more clean

Reply:
Yea might be. My way saves an allocation since it doesn't create a new `PathBuf` under the hood but it's reasonable to choose the cleaner option 👌